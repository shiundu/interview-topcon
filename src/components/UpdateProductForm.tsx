import * as React from 'react';
import { Product } from '../models';

interface Props {
    name: string,
    price: string,
    imgUrl: string,
    deleteProduct: (id: number) => void,
    handleSubmit: () => void,
    updateValue: (target: any) => void,
    cancelEdit: () => void
}

interface State {
    product: Product
}

export default class UpdatProductForm extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
  }

  render() {
        const { name, price, imgUrl, updateValue, handleSubmit, cancelEdit } = this.props;
        return ( 
            <div>
                <input
                    type="text" 
                    name="name" 
                    value={name} 
                    placeholder="Product name" 
                    onChange={e => updateValue(e)}
                />
                <input
                    type="string" 
                    name="price" 
                    value={price} 
                    placeholder="Price" 
                    onChange={e => updateValue(e)}
                />
                <input 
                    type="url" 
                    name="imgUrl" 
                    value={imgUrl} 
                    placeholder="Product image url" 
                    onChange={e => updateValue(e)} 
                />
                <div className="card-body text-center" style={{padding: '15px'}}>
                    <div className="btn-group" role="group">
                        <button
                            onClick={() => handleSubmit()}
                            className="btn btn-warning btn-sm"
                        >
                            Update
                        </button>
                        <button 
                            className="btn btn-danger btn-sm"
                            onClick={() => cancelEdit()}
                        >
                            Cancel
                        </button>
                        
                    </div>
                </div>
            </div>
        )
  }
}