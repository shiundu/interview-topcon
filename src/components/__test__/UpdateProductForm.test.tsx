import * as React from 'react';
import * as renderer from 'react-test-renderer';
import UpdateProductForm from '../UpdateProductForm';

const props = {
    name: "",
    price: "",
    imgUrl: "",
    deleteProduct: () => {},
    handleSubmit: () => {},
    updateValue: ({}) => {},
    cancelEdit: () => {}
};

describe('UpdateProductForm component test', () => {
    it('UpdateProductForm component renders correctly', () => {

        const tree = renderer.create(<UpdateProductForm {...props} />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('UpdateProductForm component has instances', async () => {

        const tree = renderer.create(<UpdateProductForm {...props} />);
        const instance = tree.getInstance();
        const render = instance.render();
        expect(render).toBeTruthy();
        
    });

})