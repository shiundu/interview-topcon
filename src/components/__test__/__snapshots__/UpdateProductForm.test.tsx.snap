// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateProductForm component test UpdateProductForm component renders correctly 1`] = `
<div>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Product name"
    type="text"
    value=""
  />
  <input
    name="price"
    onChange={[Function]}
    placeholder="Price"
    type="string"
    value=""
  />
  <input
    name="imgUrl"
    onChange={[Function]}
    placeholder="Product image url"
    type="url"
    value=""
  />
  <div
    className="card-body text-center"
    style={
      Object {
        "padding": "15px",
      }
    }
  >
    <div
      className="btn-group"
      role="group"
    >
      <button
        className="btn btn-warning btn-sm"
        onClick={[Function]}
      >
        Update
      </button>
      <button
        className="btn btn-danger btn-sm"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
